### DISCUSSION TITLE
Union type for prisma nodes

### DISCUSSION THREAD

**Discussion by P4sca1** [hayes/pothos#442] [Q&A] [answered]
How do you create a union type which consists of object types created with `t.prismaNode`?

**Comment by P4sca1** [comment]
I tried using the return types of `t.prismaNode` in the `types` array, which yields the correct schema, but throws a runtime error, that I need to either define `resolveType` on the union or `isTypeOf` on the node.
Is this something, that can be solved using the branding that pothos adds to prisma objects?

**Comment by P4sca1** [comment, marked as answer]
I found out how to do it 🥳 
You need to add `typeBrandKey` when returning the nodes, so that Pothos / GraphQL knows the correct type of the object at runtime.

```ts
import { typeBrandKey } from '@pothos/core'

const NodeA = builder.prismaNode({ ... })
const NodeB = builder.prismaNode({ ... })

const MyUnion = builder.unionType('MyUnion', { types: [NodeA, NodeB] })

builder.queryFields((t) => ({
  myUnionQuery: t.field({
    type: MyUnion,
    resolve: () => {
      // ...
      return {
        ...prismaNode,
        [typeBrandKey]: 'NodeA' // The name of the prisma model that is returned
      }
    }
})
```

**Reply by hayes** [reply]
this works, typeBrandKey is an internal symbol pothos uses when loading relay nodes.   The normal way to do this is to add a `__typename`,  this is what graphql uses by default, pothos uses `[typeBrandKey]` because it won't be as likely to interfere with existing properties.  I've seen __typename added to the DB as a column with a default value, which avoids needing to add it manually.

`NodeA.addBrand(result)` will accept either a single item or an array, and will add the branded property as a non-enumerable property.  This is how pothos does it internally.

**Reply by hayes** [reply]
relevant prisma issue: https://github.com/prisma/prisma/issues/5315

**Reply by P4sca1** [reply]
Thanks for the hint on the `addBrand(result)` helper method.
